src/modules/adyen-configuration/adyen-key-utils.ts

```diff
@@ -0,0 +1,119 @@
+import { Management, Client as AdyenClient } from "@adyen/api-library";
+import { type MeApiCredential } from "@adyen/api-library/lib/src/typings/management/meApiCredential";
+import { createLogger, logger } from "../../lib/logger";
+import { FieldError } from "../trpc/utils";
+import { type AdyenConfigurator } from "./adyen-configuration";
+
+export const InvalidPrivateKeyError = FieldError.subclass("InvalidPrivateKeyError", {
+  props: { fieldName: "apiKey" },
+});
+
+export async function getApiKeyScope(
+  apiKey: string,
+  environment: Environment,
+  configurator: AdyenConfigurator,
+): Promise<
+  | { scope: "company"; companyName: string; id: string; clientKey: string }
+  | { scope: "merchant"; companyName: string; id: string; merchantId: string; clientKey: string }
+> {
+  const logger = createLogger({ environment, saleorApiUrl: configurator.saleorApiUrl });
+  const adyenClient = new AdyenClient({
+    apiKey: apiKey,
+    environment,
+  });
+  logger.trace({ environment }, "Got Adyen Client");
+
+  const adyenManagement = new Management(adyenClient);
+  let myKey: MeApiCredential;
+  try {
+    myKey = await adyenManagement.MyAPICredentialApi.getApiCredentialDetails();
+  } catch (cause) {
+    throw new InvalidPrivateKeyError("Invalid private api key");
+  }
+
+  if (!myKey.id) {
+    throw new InvalidPrivateKeyError("API Key doesn't have an id assigned by Adyen");
+  }
+
+  if (!myKey.companyName) {
+    throw new InvalidPrivateKeyError("API key isn't linked to a company");
+  }
+
+  const companyKeyResult = await checkCompanyKey(adyenManagement, myKey.companyName, myKey.id);
+
+  if (companyKeyResult.ok) {
+    return {
+      scope: "company" as const,
+      companyName: myKey.companyName,
+      id: myKey.id,
+      clientKey: myKey.clientKey,
+    };
+  }
+
+  const clientKeyResult = await checkClientKey(adyenManagement, myKey.id);
+
+  if (clientKeyResult.ok) {
+    return {
+      scope: "merchant" as const,
+      companyName: myKey.companyName,
+      id: myKey.id,
+      merchantId: clientKeyResult.merchantId,
+      clientKey: myKey.clientKey,
+    };
+  }
+
+  throw new InvalidPrivateKeyError("API key isn't a company scoped key or merchant scoped key");
+}
+
+async function checkCompanyKey(adyenManagement: Management, companyName: string, keyId: string) {
+  try {
+    const companyScopeKey = await adyenManagement.APICredentialsCompanyLevelApi.getApiCredential(
+      companyName,
+      keyId,
+    );
+    logger.info({ companyScopeKey }, "Got company scoped key");
```

Comment:
Is `companyScopeKey` public? If not, we should change the logger level to `debug`
```suggestion
    logger.debug({ companyScopeKey }, "Got company scoped key");
```